### PR TITLE
Refactor Navigation Service Functions and Add config.json to Sample App

### PR DESCRIPTION
**Pull request by JanithGan**
## Purpose

- Move app configurations to `config.json`
- Refactor `AsgardeoNavigatorService`
- Update `README.md`

## Goals

$subject

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -153,4 +153,4 @@
         }
     },
     "defaultProject": "oidc-angular"
-}
+}
```
Comment (thivi) [inline angular.json:156]:
```suggestion
}

```

File: debug.log [hunk 2]
```diff
@@ -0,0 +1 @@
+[0105/124139.291:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
```
Comment (thivi) [inline debug.log:1]: Do we need this file?

File: lib/ng-package.json [hunk 3]
```diff
@@ -6,5 +6,6 @@
         "umdModuleIds": {
             "@asgardio/oidc-js": "oidcJs"
         }
-    }
-}
+    },
+    "whitelistedNonPeerDependencies": ["tslib", "@asgardio/oidc-js"]
+}
```
Comment (thivi) [inline lib/ng-package.json:11]:
```suggestion
}

```

File: lib/package.json [hunk 4]
```diff
@@ -1,11 +1,26 @@
...
+        "@angular/common": ">=10",
+        "@angular/core": ">=10"
     },
     "dependencies": {
-        "tslib": "^2.0.0"
+        "tslib": "^2.0.0",
+        "@asgardio/oidc-js": "^0.1.34"
     }
-}
+}
```
Comment (thivi) [inline lib/package.json:26]:
```suggestion
}

```

File: samples/basic-usage/config.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+    "signOutRedirectURL": "http://localhost:3000",
+    "clientHost": "http://localhost:3000",
+    "clockTolerance": 60,
+    "enablePKCE": true,
+    "prompt": "",
+    "scope": ["openid", "email"],
+    "storage": "sessionStorage",
+    "validateIDToken": false
+}
```
Comment (thivi) [inline samples/basic-usage/config.json:14]:
```suggestion
}

```

File: samples/basic-usage/src/app/app.component.css [hunk 6]
```diff
@@ -29,6 +29,6 @@ div.main-container {
     margin-left: 1%;
 }
 
-.mastfoot {
+.mast-foot {
     color: rgba(255, 255, 255, .5);
-}
+}
```
Comment (thivi) [inline samples/basic-usage/src/app/app.component.css:34]:
```suggestion
}

```

File: samples/basic-usage/src/app/app.component.html [hunk 7]
```diff
@@ -34,9 +32,9 @@ <h2><a routerLink="" style="color:white;text-decoration: none;">Basic Usage</a><
...
     </main>
 
-    <footer class="mastfoot mt-auto">
+    <footer class="mast-foot mt-auto">
         <div class="inner">
             <p>@2020 <a href="https://wso2.com" target="_blank" rel="noopener noreferrer">WSO2</a></p>
         </div>
     </footer>
-</div>
+</div>
```
Comment (thivi) [inline samples/basic-usage/src/app/app.component.html:40]:
```suggestion
</div>

```

File: samples/basic-usage/src/app/profile/profile.component.html [hunk 8]
```diff
@@ -32,11 +32,11 @@ <h3>{{userInfo.displayName}}</h3>
...
+                <!-- <li>Email - {{userInfo.email}}</li> -->
                 <li>Username - {{userInfo.username}}</li>
                 <li>Allowed Scopes - {{userInfo.allowedScopes}}</li>
-                <li>Tenent Domain - {{userInfo.tenantDomain}}</li>
+                <li>Tenant Domain - {{userInfo.tenantDomain}}</li>
             </div>
         </div>
     </ng-template>
-</div>
+</div>
```
Comment (thivi) [inline samples/basic-usage/src/app/profile/profile.component.html:42]:
```suggestion
</div>

```

**Review by thivi — CHANGES_REQUESTED** [review]

File: debug.log (hunk 2 above)
Comment (JanithGan) [reply debug.log:1]: no

**Review by thivi — APPROVED** [review]

**Pull request merged**